src/openrct2/ReplayManager.cpp

```diff
@@ -0,0 +1,775 @@
+            }
+
+            _nextReplayTick = gCurrentTicks + 1;
+
+            return true;
+        }
+
+    private:
+        bool ConvertDeprecatedGameCommand(const ReplayCommand& command, ReplayCommand& result)
+        {
+            // NOTE: If game actions are being ported it is required to implement temporarily
+            //       a mapping from game command to game action. This will allow the normalisation
+            //       stage to save a new replay file with the game action being used instead of the
+            //       old game command. Once normalised the code will be no longer required.
+
+            /* Example case
+            case GAME_COMMAND_RAISE_WATER:
+            {
+                uint32_t param1 = command.ebp;
+                uint32_t param2 = command.edi;
+                result.action = std::make_unique<LandRaiseWaterAction>(param1, param2, ...);
+            }
+            */
+
+            switch (command.esi)
+            {
+                case GAME_COMMAND_COUNT: // prevent default without case warning.
+                    break;
+                default:
+                    throw std::runtime_error("Deprecated game command requires replay translation.");
+            }
+
+            return true;
+        }
+
+        bool TranslateDeprecatedGameCommands(ReplayRecordData& data)
+        {
+            for (auto&& replayCommand : data.commands)
+            {
+                if (replayCommand.action == nullptr)
+                {
+                    // Check if we can create a game action with the command id.
+                    uint32_t commandId = replayCommand.esi;
+                    if (GameActions::IsValidId(commandId))
+                    {
+                        // Convert
+                        ReplayCommand converted;
+                        if (!ConvertDeprecatedGameCommand(replayCommand, converted))
+                        {
+                            return false;
+                        }
+
+                        data.commands.erase(replayCommand);
+                        data.commands.emplace(std::move(converted));
+                    }
+                }
+            }
+            return true;
+        }
+
+        bool LoadReplayDataMap(ReplayRecordData& data)
+        {
+            try
+            {
+                data.parkData.SetPosition(0);
+
+                auto context = GetContext();
+                auto& objManager = context->GetObjectManager();
+                auto importer = ParkImporter::CreateS6(context->GetObjectRepository());
+
+                auto loadResult = importer->LoadFromStream(&data.parkData, false);
+                objManager.LoadObjects(loadResult.RequiredObjects.data(), loadResult.RequiredObjects.size());
+
+                importer->Import();
+
+                sprite_position_tween_reset();
+
+                std::memcpy(gSpriteSpatialIndex, data.spriteSpatialData.GetData(), data.spriteSpatialData.GetLength());
```

Comment:
Add a check the size is correct

Reply:
Should I clamp it to the smallest value or just return false with log_error.

Reply:
Updated it, if we increase the limit the unused fields will be at least cleared now and added assert so OOB won't be possible.